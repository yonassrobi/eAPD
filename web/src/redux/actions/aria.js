export const ARIA_ANNOUNCE_CHANGE = 'ARIA_ANNOUNCE_CHANGE';

export const APD_CREATE_SUCCESS_MSG =
  'You have successfully created an APD. Close this message and continue to fill out the rest of the APD.';

export const ariaAnnounceFFPQuarterly =
  (activityId, year, q, name) => (dispatch, getState) => {
    dispatch({
      type: ARIA_ANNOUNCE_CHANGE,
      message: `${
        getState().budget.activities[activityId].quarterlyFFP.years[year][q][
          name
        ].dollars
      } dollars`
    });
  };

export const ariaAnnounceApdCreateSuccess = () => ({
  type: ARIA_ANNOUNCE_CHANGE,
  message: APD_CREATE_SUCCESS_MSG
});

export const ariaAnnounceApdLoading = () => ({
  type: ARIA_ANNOUNCE_CHANGE,
  message: 'Your APD is loading.'
});

export const ariaAnnounceApdLoaded = () => ({
  type: ARIA_ANNOUNCE_CHANGE,
  message:
    'Your APD is loaded and ready to edit. Changes to this APD will be saved automatically.'
});

export const ariaAnnounceApdLoadingFailure = error => ({
  type: ARIA_ANNOUNCE_CHANGE,
  message: error
});
