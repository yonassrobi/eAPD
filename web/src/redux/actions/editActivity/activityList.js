import { generateKey } from '@cms-eapd/common';
import {
  ADD_APD_ITEM,
  APD_ACTIVITIES_CHANGE,
  EDIT_APD,
  REMOVE_APD_ITEM
} from '../editApd';

/**
 * Add a new activity to the current APD
 */

export const addActivity = () => (dispatch, getState) => {
  const key = generateKey();
  dispatch({
    type: ADD_APD_ITEM,
    path: '/activities/-',
    state: getState(),
    key
  });
  dispatch({
    type: APD_ACTIVITIES_CHANGE,
    activities: getState().apd.data.activities,
    apdType: getState().apd.data.apdType
  });
};

/**
 * Remove an activity from the current APD
 * @param {Number} index The index of the activity to be removed
 */
export const removeActivity = index => (dispatch, getState) => {
  dispatch({ type: REMOVE_APD_ITEM, path: `/activities/${index}` });
  dispatch({
    type: APD_ACTIVITIES_CHANGE,
    activities: getState().apd.data.activities,
    apdType: getState().apd.data.apdType
  });
};

/**
 * Rename an activity
 * @param {Number} index The index of the activity to rename
 * @param {String} name The new activity name
 */
export const setActivityName = (index, name) => (dispatch, getState) => {
  dispatch({
    type: EDIT_APD,
    path: `/activities/${index}/name`,
    value: name
  });
  dispatch({
    type: APD_ACTIVITIES_CHANGE,
    activities: getState().apd.data.activities,
    apdType: getState().apd.data.apdType,
    url: `/apd/activity/${index}/overview`
  });
};

/**
 * Change an activity's Medicaid program funding source
 * @param {Number} index The index of the activity to change
 * @param {String} source The funding source to switch to - 'HIE', 'HIT', or 'MMIS'
 */
export const setActivityFundingSource = (index, source) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/activities/${index}/fundingSource`,
    value: source
  });
};
