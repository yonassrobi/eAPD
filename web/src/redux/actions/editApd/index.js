export * from './symbols';

export * from './assurancesAndCompliance';
export * from './incentivePayments';
export * from './keyPersonnel';
export * from './overview';
export * from './previousActivities';
export * from './stateProfile';
export * from './uploadFile';
export * from './securityPlanning';
