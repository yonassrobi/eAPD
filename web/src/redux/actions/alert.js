import { APD_CREATE_SUCCESS_MSG } from './aria';

export const ALERT_SUCCESS = 'ALERT_SUCCESS';
export const ALERT_RESOLVE = 'ALERT_RESOLVE';

export const alertApdCreateSuccess = id => ({
  type: ALERT_SUCCESS,
  apdId: id,
  message: APD_CREATE_SUCCESS_MSG
});

export const resolveAlertMessage = i => dispatch => {
  dispatch({
    type: ALERT_RESOLVE,
    message_index: i
  });
};
