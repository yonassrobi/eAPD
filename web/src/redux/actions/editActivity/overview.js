import { EDIT_APD } from '../editApd/symbols';

export const setActivityOverview = (activityIndex, overview) => ({
  type: EDIT_APD,
  path: `/activities/${activityIndex}/activityOverview/summary`,
  value: overview
});

export const setActivityDescription = (activityIndex, description) => ({
  type: EDIT_APD,
  path: `/activities/${activityIndex}/activityOverview/description`,
  value: description
});

export const setActivityAlternatives = (activityIndex, alternatives) => ({
  type: EDIT_APD,
  path: `/activities/${activityIndex}/activityOverview/alternatives`,
  value: alternatives
});

export const setActivityProblemStatement = (
  activityIndex,
  problemStatement
) => ({
  type: EDIT_APD,
  path: `/activities/${activityIndex}/activityOverview/problemStatement`,
  value: problemStatement
});

export const setActivityProposedSolution = (
  activityIndex,
  proposedSolution
) => ({
  type: EDIT_APD,
  path: `/activities/${activityIndex}/activityOverview/proposedSolution`,
  value: proposedSolution
});

export const setActivitySnapshot = (activityIndex, activitySnapshot) => ({
  type: EDIT_APD,
  path: `/activities/${activityIndex}/activityOverview/activitySnapshot`,
  value: activitySnapshot
});
