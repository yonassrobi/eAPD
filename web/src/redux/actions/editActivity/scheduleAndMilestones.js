import { ADD_APD_ITEM, EDIT_APD, REMOVE_APD_ITEM } from '../editApd';

/**
 * Adds or updates a milestone resource from an activity
 * @param {Number} activityIndex Index of the activity the milestone is part of
 * @param {Number} milestoneIndex Index of the milestone
 * @param {Object} data payload of the milestone to be saved
 */
export const saveMilestone =
  (activityIndex, milestoneIndex, data) => (dispatch, getState) => {
    const previousState = getState();

    let indexCalculated = milestoneIndex;

    if (
      previousState.apd.data.activities[activityIndex].milestones[
        milestoneIndex
      ] === undefined
    ) {
      indexCalculated =
        previousState.apd.data.activities[activityIndex].milestones.length;
      dispatch({
        type: ADD_APD_ITEM,
        path: `/activities/${activityIndex}/milestones/-`,
        state: getState()
      });
    }

    dispatch({
      type: EDIT_APD,
      path: `/activities/${activityIndex}/milestones/${indexCalculated}`,
      value: data
    });
  };

/**
 * Remove a milestone resource from an activity
 * @param {Number} activityIndex Index of the activity to remove the milestone from
 * @param {Number} milestoneIndex Index of the milestone to remove
 */
export const removeMilestone = (activityIndex, milestoneIndex) => dispatch => {
  dispatch({
    type: REMOVE_APD_ITEM,
    path: `/activities/${activityIndex}/milestones/${milestoneIndex}`
  });
};

export const setActivityStartDate = (activityIndex, date) => ({
  type: EDIT_APD,
  path: `/activities/${activityIndex}/activitySchedule/plannedStartDate`,
  value: date
});

export const setActivityEndDate = (activityIndex, date) => ({
  type: EDIT_APD,
  path: `/activities/${activityIndex}/activitySchedule/plannedEndDate`,
  value: date
});
