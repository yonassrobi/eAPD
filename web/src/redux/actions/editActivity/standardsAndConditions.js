import { EDIT_APD } from '../editApd/symbols';

export const setActivityStandardAndConditionDoesNotSupportExplanation = (
  activityIndex,
  explanation
) => ({
  type: EDIT_APD,
  path: `/activities/${activityIndex}/activityOverview/standardsAndConditions/doesNotSupport`,
  value: explanation
});

export const setActivityStandardAndConditionSupportExplanation = (
  activityIndex,
  explanation
) => ({
  type: EDIT_APD,
  path: `/activities/${activityIndex}/activityOverview/standardsAndConditions/supports`,
  value: explanation
});
