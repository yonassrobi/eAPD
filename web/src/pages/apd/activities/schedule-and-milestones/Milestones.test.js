import { shallow } from 'enzyme';
import React from 'react';

import {
  plain as Milestones,
  mapDispatchToProps,
  mapStateToProps
} from './Milestones';

import { removeMilestone } from '../../../../redux/actions/editActivity';

describe('the Milestones component', () => {
  const props = {
    activity: {
      milestones: [
        {
          key: 'milestone 1',
          milestone: 'Liberation Day',
          // The Netherlands is liberated from Nazi control.
          endDate: '1945-05-05'
        }
      ]
    },
    activityIndex: 7,
    remove: jest.fn()
  };

  const component = shallow(<Milestones {...props} />);

  beforeEach(() => {
    props.remove.mockClear();
  });

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  describe('events', () => {
    const list = component.find('FormAndReviewList');

    it('handles removing a milestone', () => {
      list.prop('onDeleteClick')(0);

      expect(props.remove).toHaveBeenCalledWith(7, 0);
    });
  });

  describe('redux', () => {
    it('map state to props', () => {
      const state = {
        apd: {
          data: {
            activities: ['activity 1', 'activity 2', 'activity 3']
          }
        }
      };

      expect(mapStateToProps(state, { activityIndex: 2 })).toEqual({
        activity: 'activity 3'
      });
    });

    it('map dispatch to props', () => {
      expect(mapDispatchToProps).toEqual({
        remove: removeMilestone
      });
    });
  });
});
