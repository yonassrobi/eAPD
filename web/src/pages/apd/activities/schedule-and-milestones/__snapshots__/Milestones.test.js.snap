// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Milestones component renders correctly 1`] = `
<Subsection
  headerClassName={null}
  id={null}
  nested={false}
  resource="activities.milestones"
>
  <div
    className="mb3"
  >
    <hr />
    <FormAndReviewList
      activityIndex={7}
      addButtonText="Add Milestone"
      allowDeleteAll={true}
      className={null}
      collapsed={[Function]}
      errorCheck={false}
      expanded={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "WrappedComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "activityIndex": [Function],
              "index": [Function],
              "item": [Function],
              "saveMilestone": [Function],
              "setFormValid": [Function],
            },
            "render": [Function],
          },
          "render": [Function],
        }
      }
      list={
        Array [
          Object {
            "endDate": "1945-05-05",
            "key": "milestone 1",
            "milestone": "Liberation Day",
          },
        ]
      }
      noDataMessage="Add milestone(s) for this activity."
      onAddClick={[Function]}
      onCancelClick={[Function]}
      onDeleteClick={[Function]}
    />
  </div>
</Subsection>
`;
